app/controllers/game_search_controller.rb

```diff
@@ -36,6 +29,16 @@ def create_game_objects(params)
     elsif params[:similar]
       game_name = params[:similar].to_s
       game = results.create_game_objects(params[:similar])
+    elsif params[:search_type] == 'keyword'
+      games = results.games_by_keywords(params[:search])
+      return games if games.is_a?(String)
+      session[:search_ids] = games.map(&:id) if games.is_a?(Array)
+    elsif params[:search_type] == 'quiz'
+      boxes = params.keys.find_all {|param| param.include?("box")}
+      fear_keywords = params.permit(boxes).to_h.values.join(',')
+      
+      games = results.games_by_keywords(fear_keywords)
+      session[:search_ids] = games.map(&:id) if games.is_a?(Array)
```

Comment:
were thinking this is getting a bit crowded and convoluted. A refactor to multiple controllers like `game_game_search_controller`
`keyword_game_search_controller`
`quiz_game_search_controller`

...but that's a lot of work and maybe something to talk about in the eval for areas to improve

Reply:
To me, this makes sense. The controller names can get messy like game game sounds kind of strange. Maybe something using namespace and put different aspects that depend on game within that namespace. This sounds good to mention in our things to improve on

---> game
           ---> game:search
          ----> game:keyword